src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/CustomerSegmentHandler.php

```diff
@@ -178,47 +203,64 @@ public function saveCustomerSegmentAsOrderMeta($orderId): void
         }
     }
 
-    public function showProductQtyBasedCart(): void
+    public function showProductOnProductPage(): void
     {
         global $woocommerce;
 
-        if (is_product()) {
-            $productId = get_the_ID();
-        } elseif (is_cart()) {
-            $productId = null;
+        if (!is_product()) {
+            return;
         }
 
+        $productId = get_the_ID();
         $quantity = 0;
 
-        if (is_cart()) {
-            foreach ($woocommerce->cart->get_cart() as $cartItemKey => $cartItem) {
-                if (null === $productId || $cartItem['product_id'] === $productId) {
-                    $quantity = $cartItem['quantity'];
-                    break;
-                }
-            }
-        } elseif (is_product() && $productId) {
+        if ($productId) {
             $userId = get_current_user_id();
             if ($userId) {
-                foreach ($woocommerce->cart->get_cart() as $cartItemKey => $cartItem) {
+                foreach ($woocommerce->cart->get_cart() as $cartItem) {
                     if ($cartItem['product_id'] === $productId) {
                         $quantity = $cartItem['quantity'];
                         break;
                     }
                 }
-                $segmentPrice = SegmentPriceManager::getSegmentPrice($productId, $userId, $quantity);
-                $productPrice = wc_price($segmentPrice);
+
+                if ($quantity > 0) {
+                    $segmentPrice = SegmentPriceManager::getSegmentPrice($productId, $userId, $quantity);
+                    $productPrice = wc_price($segmentPrice);
+
+                    $this->enqueueQuantityUpdateScript($quantity, $productPrice);
+                }
             }
         }
+    }
 
-        if ($quantity > 0) {
-            $jsUrl = plugins_url('storekeeper-for-woocommerce/resources/js/frontend/price.js');
+    public function showProductOnCart(): void
+    {
+        global $woocommerce;
 
-            wp_enqueue_script('quantity-update-script', $jsUrl, ['jquery'], null, true);
-            wp_localize_script('quantity-update-script', 'productData', [
-                'quantity' => $quantity,
-                'price' => $productPrice ?? 0,
-            ]);
+        if (!is_cart()) {
+            return;
         }
+
+        $quantity = 0;
+
+        foreach ($woocommerce->cart->get_cart() as $cartItem) {
+            $quantity += $cartItem['quantity'];
+        }
+
+        if ($quantity > 0) {
+            $this->enqueueQuantityUpdateScript($quantity, 0);
```

Comment:
when you have multiple products in cart how does it know which product to apply to ? 
I mean in here https://github.com/storekeeper-company/storekeeper-woocommerce-b2c/pull/151/files#diff-bfb8169259fb5ee172a482469b3d37728adca89538c10a5839da4f404e7568a9R107